### PR TITLE
Restyle unified search skeleton items animation and simplify their code

### PR DESCRIPTION
**Pull request by claucambra**
Addresses a point in #4432

New animation:

https://user-images.githubusercontent.com/70155116/178481904-1eaa1b6d-2287-41de-b830-d72c1cc622c9.mov

Old animation:

https://user-images.githubusercontent.com/70155116/178308771-e554cb1b-e088-4c81-a99a-3053e5c18f82.mov

Signed-off-by: Claudio Cambra <claudio.cambra@gmail.com>

### CODE REVIEW

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Much better – some additions:
- [ ] Position: The skeleton elements are not at the same positions at which the actual elements (icons and text) would be
- [ ] Animation: There are 2 separate animations, 1 going over the icon and 1 over the text. Can this be combined into one?
- [ ] Color: The dark grey could be lightened a bit so the difference doesn't look as harsh
- [ ] Rounding: The icon square would look much better as a circle, and the text blocks also with "pill"-style border radius so the ends are round

**Comment by claucambra** [comment]
> Much better – some additions:
> 
> * [ ]  Position: The skeleton elements are not at the same positions at which the actual elements (icons and text) would be
> * [X]  Animation: There are 2 separate animations, 1 going over the icon and 1 over the text. Can this be combined into one?
> * [X]  Color: The dark grey could be lightened a bit so the difference doesn't look as harsh
> * [X]  Rounding: The icon square would look much better as a circle, and the text blocks also with "pill"-style border radius so the ends are round


https://user-images.githubusercontent.com/70155116/178481904-1eaa1b6d-2287-41de-b830-d72c1cc622c9.mov

Regarding the position, what do you mean? Is this a question of alignment, spacing, or starting height (i.e. the section headers pushing down the real list items)? I suspect it is the last one but not sure

**Comment by jancborchardt** [comment]
> Regarding the position, what do you mean? Is this a question of alignment, spacing, or starting height (i.e. the section headers pushing down the real list items)? I suspect it is the last one but not sure

I mean that the skeleton entries are not in the place where the actual entries would be when they are loaded. :) If you send a recording (or 2 screenshots) of both the skeleton screen and the loaded list, I can show you.

**Comment by claucambra** [comment]
> > Regarding the position, what do you mean? Is this a question of alignment, spacing, or starting height (i.e. the section headers pushing down the real list items)? I suspect it is the last one but not sure
> 
> I mean that the skeleton entries are not in the place where the actual entries would be when they are loaded. :) If you send a recording (or 2 screenshots) of both the skeleton screen and the loaded list, I can show you.

Looking at the entries more closely I can see what you mean, will change

**Comment by claucambra** [comment]
@jancborchardt since we need to properly align the result list items themselves (but we shouldn't do that until #4719 is merged to prevent painful merge conflicts) I will instead align the skeleton items in accordance with point 2 in #4434, which is how the real result items will be aligned soon

**Review by allexzander — APPROVED** [review]

**Review by mgallien — COMMENTED** [review]
I have seen several issues with this implementation
The whole `UnifiedSearchResultItemSkeletonContainer` should be put into a loader to avoid having all of this loaded (including the running 30 animations) permanently
This implementation is quite heavy relying on clipping and graphical effects that may require more computational power from the GPU. Could we not directly put an animated gradient on the rounded rectangle in place of the current clipping and opacity mask approach ?
I am also concerned that the 30 animations are running all on their own without being formally synced even though we would like them to be synced to ensure visual consistence (not able to know if a delay between independent animations could occur)
I was testing with a plain old window for the system tray (not related to this PR) and noticed that the visual result degrades if we enlarge the window
Most probably you will want to have those rectangles with gradients have twice the same width that the rectangle they are overlapped on top of ? That would mean a much smaller for the smaller rectangle and most probably a bigger one for the two text placeholder rectangles

**Comment by claucambra** [comment]
> I have seen several issues with this implementation
> The whole `UnifiedSearchResultItemSkeletonContainer` should be put into a loader to avoid having all of this loaded (including the running 30 animations) permanently

This is done now

> This implementation is quite heavy relying on clipping and graphical effects that may require more computational power from the GPU. Could we not directly put an animated gradient on the rounded rectangle in place of the current clipping and opacity mask approach ?

After a lot of trying, no. It seems that `OpacityMask` does not working correctly with animations and leads to the gradient rectangle being static, so we cannot use a mask with each skeleton component and instead have to rely on the clipping approach and the independent gradient rectangles.

> I am also concerned that the 30 animations are running all on their own without being formally synced even though we would like them to be synced to ensure visual consistence (not able to know if a delay between independent animations could occur)

I mean, this is theoretically a possibility, but each animation has the exact same start point, end point, and duration, and a `Repeater` will instantiate all of its children at the same time

> I was testing with a plain old window for the system tray (not related to this PR) and noticed that the visual result degrades if we enlarge the window

This should be fixed now

> Most probably you will want to have those rectangles with gradients have twice the same width that the rectangle they are overlapped on top of ? That would mean a much smaller for the smaller rectangle and most probably a bigger one for the two text placeholder rectangles

All the rectangles need to be the same size in order to move in concert -- however this should no longer be an issue

File: src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml [hunk 1]
```diff
@@ -1,49 +1,14 @@
...
-    property int iconLeftMargin: 12
-    property int itemHeight: Style.trayWindowHeaderHeight
-    property int titleFontSize: Style.topLinePixelSize
-    property int sublineFontSize: Style.subLinePixelSize
-    property color titleColor: Style.ncTextColor
-    property color sublineColor: Style.ncSecondaryTextColor
-    property color iconColor: "#afafaf"
-
     Repeater {
         model: 10
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml]: I've changed this to calculate the number needed based on the height of the container

**Review by mgallien — APPROVED** [review]
works really great !
nice work

**Pull request merged**